indico/modules/attachments/models/folders.py

```diff
@@ -161,6 +167,8 @@ def can_view(self, user):
         """
         if not self.object.can_access(user):
             return False
+        if self.is_hidden:
```

Comment:
do this before the access check; it's much cheaper!

Reply:
done